src/caffe/solver.cpp

```diff
@@ -775,9 +776,112 @@ void AdaGradSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
   }
 }
 
+
+template <typename Dtype>
+void AdamSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
+  const vector<float>& net_params_lr = this->net_->params_lr();
+  Dtype local_rate = rate * net_params_lr[param_id];
+  const Dtype beta1 = this->param_.momentum();
+  const Dtype beta2 = this->param_.momentum2();
+
+  const int t = this->iter_ / this->param_.stepsize() +1;
```

Comment:
I did look the ADAM algorithm again, and I still believe that this value should not be divided by stepsize. The way the weighting works for the normalization (correction) should be (1-beta1^iter_), as the weighting follows a geometric series...